src/shogun2-core/src/main/java/de/terrestris/shogun2/security/access/Shogun2PermissionEvaluator.java

```diff
@@ -85,12 +91,76 @@ public boolean hasPermission(Authentication authentication,
 	}
 
 	/**
-	 * We do currently do not support/use this implementation.
+	 *
 	 */
+	@SuppressWarnings({ "rawtypes", "unchecked" })
 	@Override
 	public boolean hasPermission(Authentication authentication,
 			Serializable targetId, String targetType, Object permission) {
-		return false;
+
+		Class<?> entityClass;
+
+		try {
+			entityClass = Class.forName(targetType);
+		} catch (ClassNotFoundException e) {
+			LOG.error("Could not create class for type: " + targetType + "(" + e.getMessage() + ")");
+			return false;
+		}
+
+		// get all available DAOs from the app context
+		Collection<GenericHibernateDao> allDaos = appContext.getBeansOfType(GenericHibernateDao.class).values();
+
+		// the DAO we'll use to get the entity from the database
+		GenericHibernateDao daoToUse = null;
+
+		// we'll first try to find the exact matching DAO
+		for (GenericHibernateDao dao : allDaos) {
+			if(dao.getEntityClass().equals(entityClass)) {
+				// we found a matching DAO
+				daoToUse = dao;
+				LOG.debug("Found an exactly matching DAO for type " + entityClass);
+				break;
+			}
+		}
+
+		// if we could not find an exact match, we'll try to use the "next best"
+		// from the entity hierarchy
```

Comment:
This will possibly give you true for a class that isn't the best choice, right?

Consider:

```
A.java (With a matching DAO)
⇧
B.java (With a matching DAO)
⇧
C.java (With a matching DAO)
⇧
D.java (No DAO)
```

If entityClass is an instance of `D`, we'll drop here, as we do not have a direct dao-match. The best bet for a dao would be the `C-Dao`, but when we iterate, `isAssignableFrom` will be true for any of the parent classes (btw. their daos). We might end up with the not-well-suited A-Dao, for example.

Did I understand this correctly? Amnd if so, can `getSuperclass()` help? We could then factor ot the doa finding into a small methdi that we could call reapeatedly unto i we actually have the best bet. What do you think?